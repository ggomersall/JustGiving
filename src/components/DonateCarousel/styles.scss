
@import '../BaseStyles/variables';

$slick-arrow-color: $purple !default;
$slick-dot-color: black !default;
$slick-dot-color-active: $slick-dot-color !default;
$slick-prev-character: "\2190" !default;
$slick-next-character: "\2192" !default;
$slick-dot-character: "\2022" !default;
$slick-dot-size: 6px !default;
$slick-opacity-default: 0.75 !default;
$slick-opacity-on-hover: 1 !default;
$slick-opacity-not-active: 0.25 !default;

.donate-carousel__wrapper {
  width: 25%;
  float: left;
  margin-bottom: 15px;
  margin-left: 20px;
  border-bottom: 2px solid $grey;
}
.carousel__item {
  h2 {
    font-weight: 300;
    font-size: 40px;
    color: $purple;
    margin-bottom: 20px;
  }
  p{
    width: 75%;
    margin: 0 auto;
  }
}
.slick-slider {
  position: relative;
  display: block;
  box-sizing: border-box;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
  text-align: center;
  padding: 20px;
}
.slick-list {
  position: relative;
  overflow: hidden;
  display: block;
  margin: 0;
  padding: 0;

  &:focus {
      outline: none;
  }

  &.dragging {
      cursor: pointer;
      cursor: hand;
  }
}
.slick-slider .slick-track,
.slick-slider .slick-list {
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  -o-transform: translate3d(0, 0, 0);
  transform: translate3d(0, 0, 0);
}

.slick-track {
  position: relative;
  left: 0;
  top: 0;
  display: block;
  margin-left: auto;
  margin-right: auto;

  &:before,
  &:after {
      content: "";
      display: table;
  }

  &:after {
      clear: both;
  }

  .slick-loading & {
      visibility: hidden;
  }
}
.slick-slide {
  float: left;
  height: 100%;
  min-height: 1px;
  [dir="rtl"] & {
      float: right;
  }
  img {
      display: block;
  }
  &.slick-loading img {
      display: none;
  }

  display: none;

  &.dragging img {
      pointer-events: none;
  }

  .slick-initialized & {
      display: block;
  }

  .slick-loading & {
      visibility: hidden;
  }

  .slick-vertical & {
      display: block;
      height: auto;
      border: 1px solid transparent;
  }
}
.slick-arrow.slick-hidden {
  display: none;
}


/* Arrows */

.slick-prev,
.slick-next {
    position: absolute;
    display: block;
    height: 20px;
    width: 20px;
    line-height: 0px;
    font-size: 0px;
    cursor: pointer;
    background: transparent;
    color: transparent;
    top: 50%;
    -webkit-transform: translate(0, -50%);
    -ms-transform: translate(0, -50%);
    transform: translate(0, -50%);
    padding: 0;
    border: none;
    outline: none;
    &:hover, &:focus {
        outline: none;
        background: transparent;
        color: transparent;
        &:before {
            opacity: $slick-opacity-on-hover;
        }
    }
    &.slick-disabled:before {
        opacity: $slick-opacity-not-active;
    }
    &:before {
        font-size: 20px;
        line-height: 1;
        color: $slick-arrow-color;
        opacity: $slick-opacity-default;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
    }
}
.slick-prev {
  left: 5px;
  &:before {
      content: $slick-prev-character;
  }
}

.slick-next {
  right: 5px;
  &:before {
      content: $slick-next-character;
  }
}