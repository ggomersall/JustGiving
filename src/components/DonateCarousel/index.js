import React from 'react';
import Slider from 'react-slick';


import './styles.scss';

const DonateCarousel = ({charity: {donationDisplayAmounts}}) => {
  const settings = {
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1
  };
  
  return(
    <div className="card donate-carousel__wrapper">
      <Slider {...settings}>
        {
          donationDisplayAmounts.map((d,i) => {
            return (
              <div className="carousel__item" key={i}>
                <h2>Â£{d.amount}</h2>
                <p>{d.description}</p>
              </div>
            )
          })
        }
      </Slider>
    </div>
  )
} 
export default DonateCarousel;