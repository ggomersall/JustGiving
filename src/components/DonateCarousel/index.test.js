import React from 'react';
import { shallow } from 'enzyme';
import DonateCarousel from './index.js';

describe('Testing DonateCarousel Component', () => {

  let wrapper;

  test('it should render data correctly', () => {
    const charity = { 
      donationDisplayAmounts: [{amount: 30, description: 'lorem test'}]
    };
    wrapper = shallow(<DonateCarousel charity={charity} />);
  });

  test('Snapshot', () => expect(<DonateCarousel />).toMatchSnapshot());

});

