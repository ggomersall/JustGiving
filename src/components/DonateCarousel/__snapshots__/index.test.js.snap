// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DonateCarousel Component Snapshot 1`] = `<DonateCarousel />`;
